__tests__/FollowerButton-test.js

```diff
@@ -0,0 +1,71 @@
+import 'react-native';
+import React from 'react';
+import FollowerButton from '@screens/app/Components/FollowerButton/FollowerButton';
+import {shallow} from 'enzyme';
+import renderer from 'react-test-renderer';
+import {Chip} from 'react-native-paper';
+
+/** Snapshot test */
+test('Follower Button Snapshot Test', () => {
+  const tree = renderer.create(<FollowerButton />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+
+/** This is currently testing the dummy placeholder code. However,
+ * it will need to be resturcutred to open up a new page that passes through an array of followers.
+ */
+function testOnFollowerClicked(user_id) {
+  it('onFollowerClicked username is returned', () => {
+    const wrapper = shallow(<FollowerButton user_id={user_id} />);
+    /** DONT FORGET TO PASS IN THE PROP TO THE FUNCTION. */
+    /** After another hour.....
+     * 1. this link helped: https://docs.reactnativestarter.com/testing
+     * 2. Import the component + search by component.
+     * Passing in String component name DOES NOT return ANY nodes.
+     */
+    const chipWrapper = wrapper.find(Chip);
+    const chipProps = chipWrapper.first().props();
+    const chipPressFunctionVal = chipProps.onPress();
+    console.log(chipPressFunctionVal);
+    expect(chipPressFunctionVal).toEqual(
+      'This is the follower list of ' + user_id,
+    );
+  });
+}
+
+/** Check that the property values passed into the prop, appear on the prop. */
+function testProperPropsPassed(user_id, follower_count) {
+  it('User ID value is the FollowerButton prop value', () => {
+    /** Does it matter if we pass in follower_count simulatanously? */
+    const wrapper = shallow(<FollowerButton user_id={user_id} />);
+    const instance = wrapper.instance();
+    const userIDProp = instance.props.user_id;
+    expect(user_id).toEqual(userIDProp);
+  });
+
+  it('Follower Count value is the FollowerButton prop value', () => {
+    /** Does it matter if we pass in follower_count simulatanously? */
+    const wrapper = shallow(<FollowerButton follower_count={follower_count} />);
```

Comment:
What if this is a negative value? Does anything in the UI change? We should make sure min is 0

Reply:
@kartikeya01  The view will be 0, as seen in testChipInnerText. However, to keep the component pure we can't change the actual prop value. If we wanted to pass that prop value over, we can use the 0 declared as the variable

Reply:
that being noted, a warning is development is noted if a nonnegative integer prop is passed in